File: C++/Easy/Accepted/greatest-english-letter-in-upper-and-lower-case_5.cpp
class Solution {
public:
    string greatestLetter(string s) {
        
        string res = "";
        int upper[26] = {0};
        int lower[26] = {0};
        for (char c : s) {
            if (isupper(c)) {
                upper[c - 'A'] = 1;
            } else {
                lower[c - 'a'] = 1;
            }
        }
        for (int i = 25; i >= 0; i--) {
            if (upper[i] == 1 && lower[i] == 1) {
                res += (i + 'A');
                return res;
            }
        }
        return res;
    }
};

File: Java/Easy/RuntimeError/greatest-english-letter-in-upper-and-lower-case_5.java
class Solution {
    public String greatestLetter(String s) {
        
        int[] counts = new int[26];
        for (char c : s.toCharArray()) {
            counts[c - 'a']++;
        }
        for (int i = 25; i >= 0; i--) {
            if (counts[i] > 0 && counts[i + 26] > 0) {
                return String.valueOf((char) ('A' + i));
            }
        }
        return "";
    }
}